src/collection.ts

```diff
@@ -126,7 +126,7 @@ export interface CollectionOptions
 
 /** @internal */
 export interface CollectionPrivate {
-  pkFactory: PkFactory | typeof ObjectId;
+  pkFactory: PkFactory;
```

Comment:
Our ObjectId class does have a `createPk` method it's just missing from the community types so I prefer here to cast in the constructor which we can revisit when we have correct typing.

Reply:
Is this because ObjectID shouldn't have createPk?
cus wouldn't this just be:
```
class DefaultPkFactory {
  static createPk() { return ObjectId.createPk() }
}
```
Why the pass through class?